contracts/KyberOasisReserve.sol

```diff
@@ -0,0 +1,273 @@
+pragma solidity 0.4.18;
+
+import "./ERC20Interface.sol";
+import "./Utils2.sol";
+import "./Withdrawable.sol";
+import "./KyberReserveInterface.sol";
+
+
+contract OtcInterface {
+    function getBuyAmount(ERC20 buyGem, ERC20 payGem, uint payAmt) public constant returns (uint fillAmt);
+    function sellAllAmount(ERC20 payGem, uint payAmt, ERC20 buyGem, uint minFillAmount) public returns (uint fillAmt);
+}
+
+
+contract TokenInterface is ERC20 {
+    function deposit() public payable;
+    function withdraw(uint) public;
+}
+
+
+contract KyberOasisReserve is KyberReserveInterface, Withdrawable, Utils2 {
+
+    uint constant internal ETH_SRC_AMOUNT = (10**18);
+    address public kyberNetwork;
+    OtcInterface public otc;
+    TokenInterface public wethToken;
+    ERC20 public tradeToken;
+    bool public tradeEnabled;
+    uint public feeBps;
+
+    function KyberOasisReserve(
+        address _kyberNetwork,
+        OtcInterface _otc,
+        TokenInterface _wethToken,
+        ERC20 _tradeToken,
+        address _admin,
+        uint _feeBps
+    ) public {
+        require(_admin != address(0));
+        require(_kyberNetwork != address(0));
+        require(_otc != address(0));
+        require(_wethToken != address(0));
+        require(_tradeToken != address(0));
+        require(_feeBps < 10000);
+        require(getDecimals(_wethToken) == MAX_DECIMALS);
+        require(getDecimals(_tradeToken) == MAX_DECIMALS);
```

Comment:
also here, maybe use 18, or ETH_DECIMALS, but max decimals is wrong.

Reply:
done.